lib/client/doc.js

```diff
@@ -200,7 +200,14 @@ Doc.prototype.whenNothingPending = function(callback) {
     this.once('nothing pending', callback);
     return;
   }
-  callback();
+  var doc = this;
+  process.nextTick(function() {
+    if (doc.hasPending()) {
```

Comment:
There is duplicated logic here, which it seems is not required.

Your current formulation is this:

```js
Doc.prototype.whenNothingPending = function(callback) {
  if (this.hasPending()) {
    this.once('nothing pending', callback);
    return;
  }
  var doc = this;
  process.nextTick(function() {
    if (doc.hasPending()) {
      doc.once('nothing pending', callback);
      return;
    }
    callback();
  });
};
```

but could it be this instead?

```js
Doc.prototype.whenNothingPending = function(callback) {
  var doc = this;
  process.nextTick(function() {
    if (doc.hasPending()) {
      doc.once('nothing pending', callback);
      return;
    }
    callback();
  });
};
```

The end result should be the same in terms of behavior.

Reply:
Thanks @curran , I've just made this change.